doc/modules/mod_jingle_sip.md

```diff
@@ -1,26 +1,88 @@
 
-* `INVITE` - with additionall callback for following response codes:
-   * `200`
-   * `180` and `183`
-   * `486` when the call's recipient rejects it
-   * from `400` to `600` - other error codes indicating session terminate
-* `re-INVITE` - `INVITE` message sent for established session
+* `INVITE`
+* `re-INVITE` - `INVITE` message sent for an accepted session
 * `CANCEL`
 * `BYE`
 * `INFO`
 
+and following responses to the INVITE request:
+
+* `200` when the call invite was accepted
+* `180` and `183` to indicate that the invitation was sent to the device
+* `486` when the call's recipient rejects it
+* from `400` to `600` - other error codes indicating session termination
+
+#### Jingle to SIP translation
+
+The table below summarises the translation both ways for standard Jingle and SIP messages:
+
+| Jingle action | SIP message | comment |
+| ------------- | ----------- | ------- |
+| `session-initiate` | `INVITE` request | |
+| `session-accept` | `200 OK` response | |
+| `session-terminate` with reason `success` | `BYE` request | Only for accepted session |
+| `session-terminate` with reason `decline` | `CANCEL` request | When sent by call's initiator |
+| `session-terminate` with reason `decline` | `486 Busy Here` response | When sent by the invite user |
+| `transport-info` | `INFO` request | |
+
+##### Ringing notification
+
+Both Jingle and SIP has the `ringing` notification.
+It's generated as a response code `180 Ringing` by a SIP entity when the INVITE is sent to the device.
+In SIP world there is also `183 Session Progress` response code generated in some cases.
+Both the codes `180` and `183` are translated as `session-info` Jingle stanza with `ringing` sub element.
+MongooseIM generates only `180 Ringing` response code the `INVITE` request, if the recipient's online.
+
+##### Recipient unavailable
+
+When MongooseIM receives and SIP `INVITE` request to a user who is offline,
+it will reply with `480 Temporarily Unavailable` code.
+The same code is expected from the SIP Proxy when MongooseIM sends the `INVITE` request.
+
+##### Other error codes
+
+When MongooseIM receives any error code to the INVITE request from range `400` to `699` but other then `486`,
+it will send Jingle `session-terminate` stanza to the call's initiator.
+The stanza has reason `general-error` with the SIP error code in the `sip-error` sub element.
+
+##### Non-standard Jingle stanzas used by jingle.js
+
+The following non-standard Jingle stanzas were integrated with https://github.com/softwarehutpl/jingle.js
+
+* `source-remove`
+* `source-add`
+* `source-update`
+
+When MongooseIM observes the above Jingle stanzas it will translate them to a SIP in-dialog INVITE request.
+In the SDP content of the request, there will be a custom attribute `a=jingle-action`.
+The value of the custom attribute is one of the three presented above.
+
+Similarly when MongooseIM gets SIP in-dialog `INVITE` request it will check if there is the custom attribute
```

Comment:
Similarly when MongooseIM gets a SIP in-dialog `INVITE` request, it will check if there is a custom attribute and use it as the `action` attribute of the Jingle stanza sent to the user?